apps/src/templates/census2017/CensusMap.jsx

```diff
@@ -0,0 +1,345 @@
+/* global google adjustScroll*/
+import $ from 'jquery';
+import React, { PropTypes, Component } from 'react';
+import ReactDOM from 'react-dom';
+
+class CensusMapInfoWindow extends Component {
+  static propTypes = {
+    handleTakeSurveyClick: PropTypes.func.isRequired,
+    schoolId: PropTypes.string.isRequired,
+    schoolName: PropTypes.string.isRequired,
+    city: PropTypes.string.isRequired,
+    state: PropTypes.string.isRequired,
+    location: PropTypes.string.isRequired,
+    teachesCs: PropTypes.string.isRequired
+  };
+
+  render() {
+    let censusMessage;
+
+    switch (this.props.teachesCs) {
+      case 'YES':
+        censusMessage = "We believe this school offers Computer Science.";
+        break;
+      case 'NO':
+        censusMessage = "We believe this school offers limited or no Computer Science opportunities.";
+        break;
+      case 'MAYBE':
+        censusMessage = "We have conflicting data for this school.";
+        break;
+      default:
+        censusMessage = "We need data for this school.";
+    }
+
+    const schoolDropdownOption = {
+      value: this.props.schoolId,
+      label: `${this.props.schoolName} - ${this.props.city}, ${this.props.state}`,
+      school: {
+        nces_id: this.props.schoolId,
+        name: this.props.schoolName,
+        city: this.props.city,
+        state: this.props.state,
+        latitude: this.props.location.split(',')[0],
+        longitude: this.props.location.split(',')[1],
+      },
+    };
+
+    return (
+      <div className="census-info-window">
+        <h4>
+          <b>
+            {this.props.schoolName}
+          </b>
+          <br />
+          ({this.props.city}, {this.props.state})
+        </h4>
+        <hr />
+        <div className="census-message">{censusMessage}</div>
+        <div className="button-container">
+          <div className="button-link-div">
+            <a onClick={() => this.props.handleTakeSurveyClick(schoolDropdownOption)}>
+              <div className="button">
+                <div className="button-text">Take the survey for this school</div>
+              </div>
+            </a>
+          </div>
+          <div className="button-link-div">
+            <a href="/yourschool/letter" target="_blank">
+              <div className="button">
+                <div className="button-text">Send the survey to a teacher at this school</div>
+              </div>
+            </a>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default class CensusMap extends Component {
```

Comment:
Not ship blocking, but I'm used to seeing each component in their own file. Maybe separate in a follow up? Ideally, we'd have tests and Stories for the new component, too.

Reply:
Since these competes depend on the google maps scripts being loaded on the page I wasn't able to get stories working for them. I actually needed to set `hideMap` on the `YourSchool` story so that it wouldn't be broken. Do you know the right way to include scripts within a story?